tasks/main.yml

```diff
@@ -30,22 +30,22 @@
   become: "{{ passwordless_ssh_become }}"
   become_user: "{{ passwordless_ssh_become_user }}"
 
-- name: update known hosts with the other hosts in this current group
-  known_hosts:
-    path: "{{ passwordless_ssh_user_home }}/.ssh/known_hosts"
-    name: "{{ hostvars[item]['ansible_host'] | default(item) }}"
-    key: "{{ hostvars[item]['ansible_host'] | default(item)}} {{ _public_key }}"
-  with_items: "{{ play_hosts }}"
-  become: "{{ passwordless_ssh_become }}"
-  become_user: "{{ passwordless_ssh_become_user }}"
-  changed_when: false
+- block:
```

Comment:
I'm not so sure that this block should *replace* the above block, but rather should add to it. If I try to ssh as user `ghelling` from my localhost to a remote session, I need my user's key to be added to the `.ssh/known_hosts` file of the remote system, and not my laptop's /etc/ssh/ key. That allows me to login as my user to that remote user without password.

Does this change allow any user on the current system to login to the remote system without a password? I've never before touched nor needed to touch anything under `/etc/ssh` so I'm not sure what removing my change and adding yours will accomplish.

Reply:
This change does nothing with the `authorized_keys` handling before it, and does nothing to disallow the ability for a user to log in to a remote system without a password.

`known_hosts` has nothing to do with users, login, or the host you're connecting from. `known_hosts` is a list of host keys (never user pubkeys) that SSH uses to authenticate *hosts* that you're connecting to before user authentication begins. You do not need your user's key to be added to `known_hosts` at any time.